src/config.js

```diff
@@ -0,0 +1,1602 @@
+  constructor (param = {}) {
+    const {notificationManager, enablePersistence} = param
+    this.notificationManager = notificationManager
+    this.enablePersistence = enablePersistence
+    this.clear()
+  }
+
+  initialize ({configDirPath, resourcePath, projectHomeSchema}) {
+    this.configDirPath = configDirPath
+    this.resourcePath = resourcePath
+    if (this.enablePersistence != null) {
+      this.configFilePath = fs.resolve(this.configDirPath, 'config', ['json', 'cson'])
+      if (this.configFilePath == null) { this.configFilePath = path.join(this.configDirPath, 'config.cson') }
+    }
+
+    this.schema.properties.core.properties.projectHome = projectHomeSchema
+    this.defaultSettings.core.projectHome = projectHomeSchema.default
+  }
+
+  clear () {
+    this.emitter = new Emitter()
+    this.schema = {
+      type: 'object',
+      properties: {}
+    }
+    this.defaultSettings = {}
+    this.settings = {}
+    this.scopedSettingsStore = new ScopedPropertyStore()
+
+    this.settingsLoaded = false
+    this.savePending = false
+    this.configFileHasErrors = false
+    this.transactDepth = 0
+    this.pendingOperations = []
+    this.legacyScopeAliases = {}
+
+    this.requestLoad = _.debounce(() => {
+      this.loadUserConfig()
+    }
+      , 100)
+
+    const debouncedSave = _.debounce(() => {
+      this.savePending = false
+      this.save()
+    }
+      , 100)
+    this.requestSave = () => {
+      this.savePending = true
+      debouncedSave()
+    }
+  }
+
+  shouldNotAccessFileSystem () { return !this.enablePersistence }
+
+  /*
+  Section: Config Subscription
+  */
+
+  // Essential: Add a listener for changes to a given key path. This is different
+  // than {::onDidChange} in that it will immediately call your callback with the
+  // current value of the config entry.
+  //
+  // ### Examples
+  //
+  // You might want to be notified when the themes change. We'll watch
+  // `core.themes` for changes
+  //
+  // ```coffee
+  // atom.config.observe 'core.themes', (value) ->
+  //   # do stuff with value
+  // ```
+  //
+  // * `keyPath` {String} name of the key to observe
+  // * `options` (optional) {Object}
+  //   * `scope` (optional) {ScopeDescriptor} describing a path from
+  //     the root of the syntax tree to a token. Get one by calling
+  //     {editor.getLastCursor().getScopeDescriptor()}. See {::get} for examples.
+  //     See [the scopes docs](http://flight-manual.atom.io/behind-atom/sections/scoped-settings-scopes-and-scope-descriptors/)
+  //     for more information.
+  // * `callback` {Function} to call when the value of the key changes.
+  //   * `value` the new value of the key
+  //
+  // Returns a {Disposable} with the following keys on which you can call
+  // `.dispose()` to unsubscribe.
+  observe () {
+    let callback, keyPath, options, scopeDescriptor
+    if (arguments.length === 2) {
+      [keyPath, callback] = Array.from(arguments)
```

Comment:
you don't need `Array.from()` here

Reply:
I'd even say rest params would be even better here